community/kernel/src/main/java/org/neo4j/graphdb/impl/notification/NotificationCode.java

```diff
@@ -79,8 +79,8 @@
         Status.Statement.DynamicPropertyWarning,
         "Using a dynamic property makes it impossible to use an index lookup for this query"
     ),
-    BARE_NODE_SYNTAX_DEPRECATED(
-        SeverityLevel.WARNING,
+    BARE_NODE_SYNTAX_DEPRECATED( // This notification is no longer produced by current Cypher compilers
+        SeverityLevel.WARNING,   // but it is left here for backwards compatibility.
```

Comment:
Not quite. Kernel has no dependencies on older compiler directly, but Cypher 3.0 will depend on compilers that can send out this notification, and so we can't remove it from the 3.0 kernel. Not sure a better alternative exists.